packages/web/components/Player/Section/PlayerMeToken.tsx

```diff
@@ -0,0 +1,648 @@
+import {
+  Box,
+  Button,
+  Flex,
+  IconButton,
+  Image,
+  Input,
+  MetaButton,
+  Tab,
+  TabList,
+  TabPanel,
+  TabPanels,
+  Tabs,
+  Text,
+  useToast,
+  Wrap,
+} from '@metafam/ds';
+import { ProfileSection } from 'components/Section/ProfileSection';
+import { ethers } from 'ethers';
+import { Player } from 'graphql/autogen/types';
+import { useWeb3 } from 'lib/hooks';
+import React, { useEffect, useState } from 'react';
+import { HiOutlineInformationCircle, HiSwitchVertical } from 'react-icons/hi';
+import { BoxTypes } from 'utils/boxTypes';
+import { switchChainOnMetaMask } from 'utils/metamask';
+import {
+  approveMeTokens,
+  burn,
+  getCollateralData,
+  getErc20TokenData,
+  getMeTokenFor,
+  getMeTokenInfo,
+  mint,
+  nullMeToken,
+  preview,
+  spendMeTokens,
+} from 'utils/meTokens';
+
+type TokenData = {
+  collateral: string;
+  profilePicture: string;
+  symbol: string;
+  tokenAddress: string;
+};
+
+type Props = {
+  player: Player;
+  isOwnProfile?: boolean;
+  editing?: boolean;
+};
+
+type BlockProps = {
+  symbol: string;
+  profilePicture: string;
+  address: string;
+};
+
+type SwapProps = {
+  symbol: string;
+  profilePicture: string;
+  collateral: any;
+  metokenAddress: string;
+  owner: string;
+  provider: any;
+};
+
+type LiveCollateralData = {
+  image: string;
+  currentPrice: string;
+};
+
+const MeTokenSwap: React.FC<SwapProps> = ({
+  symbol,
+  profilePicture,
+  metokenAddress,
+  collateral,
+  owner,
+  provider,
+}) => {
+  const [liveCollateralData, setLiveCollateralData] =
+    useState<LiveCollateralData>();
+  const [collateralTokenData, setCollateralTokenData] = useState<any>();
+  const [meTokenData, setMeTokenData] = useState<any>();
+  const [transactionType, toggleTransactionType] = useState<string>('mint');
+  const [approved, setApproved] = useState<boolean>(false);
+  const { chainId, address } = useWeb3();
+  const [amount, setAmount] = useState<string>('0');
+  const [previewAmount, setPreviewAmount] = useState<string>('0');
+  const [loading, setLoading] = useState<boolean>(false);
+  const toast = useToast();
+
+  useEffect(() => {
+    if (!collateral) return;
+    const getLiveData = async () => {
+      setLiveCollateralData(await getCollateralData(collateral));
+    };
+    getLiveData();
+  }, [collateral]);
+
+  useEffect(() => {
+    if (!metokenAddress || !collateral) return;
+    const getInAndOutTokenData = async () => {
+      await getErc20TokenData(collateral, owner).then((res) => {
+        setCollateralTokenData(res);
+      });
+      await getErc20TokenData(metokenAddress, owner).then((res) => {
+        setMeTokenData(res);
+      });
+    };
+    getInAndOutTokenData();
+  }, [metokenAddress, collateral]); // eslint-disable-line react-hooks/exhaustive-deps
+
+  useEffect(() => {
+    if (!amount || !metokenAddress) return;
+    handlePreview();
+  }, [transactionType, amount, metokenAddress]); // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
Ditto, handlePreview itself has a bunch of dependencies

Reply:
Moved this to a useCallback